rogue-thi-app/lib/hooks/user-kind.js

```diff
@@ -1,21 +1,33 @@
 import { useEffect, useState } from 'react'
+import API from '../../lib/backend/authenticated-api'
 
 export const USER_STUDENT = 'student'
 export const USER_EMPLOYEE = 'employee'
 export const USER_GUEST = 'guest'
 
 export function useUserKind () {
   const [userKind, setUserKind] = useState(USER_GUEST)
+  const [userFaculty, setUserFaculty] = useState(null)
 
   useEffect(() => {
-    if (localStorage.session === 'guest2') {
-      setUserKind(USER_GUEST)
-    } else if (localStorage.isStudent === 'false') {
-      setUserKind(USER_EMPLOYEE)
-    } else {
-      setUserKind(USER_STUDENT)
+    async function loadFaculty () {
+      // return 'Nachhaltige Infrastruktur'
+      return await API.getFaculty()
     }
+
+    async function load () {
+      if (localStorage.session === 'guest2') {
+        setUserKind(USER_GUEST)
+      } else if (localStorage.isStudent === 'false') {
+        setUserKind(USER_EMPLOYEE)
+      } else {
+        setUserKind(USER_STUDENT)
+        setUserFaculty(await loadFaculty())
```

Comment:
Why don't we also store the faculty in LocalStorage? With your implementation we have an unnecessary delay and API call when opening the map.